### DISCUSSION TITLE
How do we use values from `nuxtConfig.public.ENV_VAR`?

### DISCUSSION THREAD

**Discussion by paambaati** [Baroshem/nuxt-security#542] [Q&A] [answered]
We'd like to set up some values inside `nuxt-security` configuration that will come from `nuxtConfig.public.ENV_VAR` (as we can override them during runtime with `NUXT_PUBLIC_ENV_VAR`).

How do I do that with the `nuxt-security` module?

**Comment by Baroshem** [comment, marked as answer]
Hey Buddy,

You can check out this documentation link https://nuxt-security.vercel.app/documentation/advanced/hooks#examples

But keep in mind that it does not work for evey feature (i.e basic auth cannot be configured like that)

**Reply by paambaati** [reply]
Thank you so much for the quick response.

I was to able to fix this by writing this plugin –

```typescript
// server/plugins/security.ts

import { defu } from 'defu';

export default defineNitroPlugin((nitroApp) => {
  const appBaseUrl = useRuntimeConfig().public.APP_BASE_URL;
  nitroApp.hooks.hook('nuxt-security:routeRules', (appSecurityOptions) => {
    appSecurityOptions['/**'] = defu({
      headers: {
        contentSecurityPolicy: {
          'connect-src': [appBaseUrl, appBaseUrl.replace(/^https?:\/\//, 'ws://')],
        },
      },
    });
  });
});
```
